src/expr/src/record_iter.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright Materialize, Inc. All rights reserved.
```

Comment:
I wonder if everything in this file makes more sense as a file in the `repr` crate given that its the persisted file representation of our rows / datums? wdyt justin? i also think having all of that in one crate makes it nicer to test